dev_guide/builds.adoc

```diff
@@ -1184,6 +1190,80 @@ spec:
 ----
 ====
 
+[[build-hooks]]
+
+== Build Hooks
+Build hooks can be used as a part of the build flow to run tests inside a temporary container and validate your image before it is being pushed to the registry. The command is executed with the current working directory set to the image's `WORKDIR`.
+The hook will fail if the script or command returns a non-zero exit code or if there's any other error related to starting the temporary container. When the hook fails it will mark the build as failed.
+====
+----
+spec:
+  postCommit: <1>
+    command:
+      -"sh"
+      -"-c"
+    args: "rake test" <2>
+    script: "rake test" <3>
+----
+<1> `command` is the command to run. This might be needed if the image doesn't have "/bin/sh" or if you do not want to use a shell. In all other cases, using `script` might be more convenient.
+
+<2> `args` is a list of arguments that are provided to either `command`,
+`script` or the default entrypoint of the image. The arguments are
+placed immediately after the command to be run.
+
+<3> `script` is a shell script to be run with `/bin/sh -c`. Use `script` when a shell script is appropriate to execute the post build hook, for example for running unit tests with "rake test". If you need control over the image entrypoint, or if the image does not have "/bin/sh", use `command` and/or `args`.
+
+*NOTE: It is invalid to provide both `Script` and `Command` simultaneously. If none of the fields are specified, the hook is not executed.
```

Comment:
There is some special syntax for note blocks. For a one-liner note, the case here, it is:

```asciidoc
NOTE: this is a note.
```

Note that there should be no "*" in the beginning of the line.
And if the note is lengthier, there's this format as well:

```asciidoc
[NOTE]
====
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec a diam lectus.
Sed sit amet ipsum mauris. Maecenas congue ligula ac quam viverra nec
consectetur ante hendrerit. Donec et mollis dolor. Praesent et diam eget libero
egestas mattis sit amet vitae augue.
====
```